src/main/java/org/mule/tools/apikit/input/parsers/APIAutodiscoveryConfigParser.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.tools.apikit.input.parsers;
+
+import org.jdom2.Attribute;
+import org.jdom2.Document;
+import org.jdom2.Element;
+import org.jdom2.filter.Filters;
+import org.jdom2.xpath.XPathExpression;
+import org.jdom2.xpath.XPathFactory;
+
+import org.mule.tools.apikit.model.APIAutodiscoveryConfig;
+
+import java.util.List;
+import java.util.Optional;
+
+public class APIAutodiscoveryConfigParser implements MuleConfigFileParser<APIAutodiscoveryConfig> {
+
+  private static final XPathExpression<Element> API_AUTODISCOVERY_EXPRESSION = getCompiledExpression();
+
+  private static XPathExpression<Element> getCompiledExpression() {
```

Comment:
why a method here ? 
  private static final XPathExpression<Element> API_AUTODISCOVERY_EXPRESSION =
      XPathFactory.instance()
          .compile("//*/*[local-name()='" + APIAutodiscoveryConfig.ELEMENT_NAME + "']",
              element(APIAutodiscoveryConfig.API_AUTODISCOVERY_NAMESPACE.getNamespace()));

Reply:
This was taken from APIKitConfigParser, will do refactor.